src/fsharp/NameResolution.fs

```diff
@@ -2253,7 +2253,14 @@ let rec ResolveLongIdentInTypePrim (ncenv: NameResolver) nenv lookupKind (resInf
                         addToBuffer uc.DisplayName
             | _ -> ()
 
-        raze (UndefinedName (depth, FSComp.SR.undefinedNameFieldConstructorOrMember, id, suggestMembers))
+        let errorTextF s =
+            if isAppTy g ty then
```

Comment:
Before this, check if it's a byref type (`isByrefTy`) and if it is, call `destByrefTy` to get the actual type.

Reply:
That approach I mentioned above is fine. I don't know how often we will be doing things similar to this, but it may be a good idea to add a `stripByrefTy` that will always grab the underlying type if it's a `byref`.

You can decide either approach. If you wish to add a `stripByrefTy`, you will need to rename an existing `stripByrefTy` function in `infos.fs`. That function does not exactly do what I mentioned here and is only used for determining method overloading; so that one should probably be renamed to `mapInrefAndOutrefTyToByrefTy` as it's super specific and does exactly that.